renderdoc/driver/vulkan/vk_acceleration_structure.h

```diff
@@ -37,7 +43,18 @@ class VulkanAccelerationStructureManager
     bool isTLAS;
   };
 
-  VulkanAccelerationStructureManager(WrappedVulkan *driver) : m_pDriver(driver) {}
+  explicit VulkanAccelerationStructureManager(WrappedVulkan *driver);
+
+  // ResourceId.  BDA isn't known at creation time and the usage flags aren't known at bind time so
+  // initialise is a two-step process.  These should only be called when capturing
+  void TrackInputBuffer(VkDevice device, VkResourceRecord *record);
+  void UntrackInputBuffer(VkResourceRecord *record);
+
+  // Allocates readback mem and injects commands into the command buffer so that the iput buffers
+  // are copied.
+  void CopyInputBuffers(VkCommandBuffer commandBuffer,
+                        const VkAccelerationStructureBuildGeometryInfoKHR &info,
+                        const VkAccelerationStructureBuildRangeInfoKHR *buildRange);
```

Comment:
This function is unused/orphaned in this PR so best to leave it out for now.

Reply:
It is used but only to provide the built flag in the AS' `VkResourceRecord`, which is then set true upon queue submission.  Yes it's a bit arbitrary now but it lays down the architecture for the subsequent patches.

Reply:
Oh I see, this should be done differently - the cold data (`accelerationStructureInfo` in this case) should be initialised at creation time, so in `WrappedVulkan::vkCreateAccelerationStructureKHR` where the resource record is created. That would mean this function should still be deferred to a later patch.